servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/ServiceTalkContentEncodingTest.java

```diff
@@ -123,6 +126,11 @@ protected Single<StreamingHttpResponse> request(final StreamingHttpRequester del
                 // content encoding should be stripped by the time the decoding is done.
                 assertThat(response.headers().get(CONTENT_ENCODING), nullValue());
 
+                final CharSequence varyHeader = response.headers().get(VARY);
+                if (varyHeader != null) {
+                    assertEquals(ACCEPT_ENCODING.toString(), varyHeader.toString());
```

Comment:
Just FYI that we have `io.servicetalk.buffer.api.Matchers.contentEqualTo` that can be used with hamcrest's `assertThat` to avoid null check and `toString()` enforcement

Reply:
```java
assertThat(varyHeader, contentEqualTo(ACCEPT_ENCODING));
```

Reply:
Thanks for the info. In this case we can't remove the null check since there might be cases where the vary header is null (no server compression). Should we keep it as is or should I refactor the code to check on a different if condition and then use `contentEqualTo` inside?